src/main/ruby/core/exception.rb

```diff
@@ -390,37 +401,44 @@ class KeyError < IndexError
 
 class SignalException < Exception
 
+  alias :signm :message
   attr_reader :signo
-  attr_reader :signm
-
-  def initialize(signo = nil, signm = nil)
-    # MRI overrides this behavior just for SignalException itself
-    # but not for anything that inherits from it, therefore we
-    # need this ugly check to make sure it works as intended.
-    return super(signo) unless self.class == SignalException
-    if signo.is_a? Integer
-      unless @signm = Signal::Numbers[signo]
-        raise ArgumentError, "invalid signal number #{signo}"
+
+  def initialize(sig, message = undefined)
+    signo = Rubinius::Type.rb_check_to_integer(sig, :to_int)
+    if signo.nil?
+      raise ArgumentError, 'wrong number of arguments (given 2, expected 1)' unless undefined.equal?(message)
+      if sig.is_a?(Symbol)
+        sig = sig.to_s
+      else
+        sig = StringValue(sig)
       end
-      @signo = signo
-      @signm = signm || "SIG#{@signm}"
-    elsif signo
-      if signm
-        raise ArgumentError, 'wrong number of arguments (2 for 1)'
+      signm = sig
```

Comment:
Could you give a meaningful name to this variable, like `signame` or `name` or `signal_name`?

Reply:
Yes, updated to `signal_name`